src/app/compare/compare.component.js

```diff
@@ -60,12 +61,24 @@ function CompareCtrl($scope, toaster, dimCompareService, dimStoreService, D2Stor
         }
       });
       vm.statsMap[statId] = itemStatsMap[statId].index;
-      vm.comparisons[0].stats.splice(vm.statsMap[statId], 0, { value: undefined, id: Number(statId), statHash: Number(statId), name: itemStatsMap[statId].name });
-    });
 
+      const missingStatItemIdx = vm.comparisons.findIndex((compItem) => !compItem.stats.some((stat) => stat.id === Number(statId)));
+      if (missingStatItemIdx > 0) {
+        vm.comparisons[missingStatItemIdx].stats.splice(vm.statsMap[statId], 0, { value: undefined, id: Number(statId), statHash: Number(statId), name: itemStatsMap[statId].name });
```

Comment:
The object argument should probably be split across several lines, with one line per key.